internals/daemon/api_notices.go

```diff
@@ -267,19 +249,22 @@ func v1GetNotice(c *Command, r *http.Request, _ *UserState) Response {
 	if notice == nil {
 		return NotFound("cannot find notice with ID %q", noticeID)
 	}
-	if !noticeViewableByUser(notice, requestUID, daemonUID) {
+	if !noticeViewableByUser(notice, user) {
 		return Forbidden("not allowed to access notice with id %q", noticeID)
 	}
 	return SyncResponse(notice)
 }
 
-func noticeViewableByUser(notice *state.Notice, requestUID, daemonUID uint32) bool {
+func noticeViewableByUser(notice *state.Notice, user *UserState) bool {
 	userID, isSet := notice.UserID()
 	if !isSet {
+		// Notice has no UID, so it's viewable by any user.
 		return true
 	}
-	if isAdmin(requestUID, daemonUID) {
+	if user.Access == state.AdminAccess {
```

Comment:
This is now a bit clunky. An admin user who (by some future means) doesn't have a UID will not be able to read all notices.

Reply:
True. This is per the design of notices being very much UID-based (which was arguably short-sighted?). I've added a TODO comment in https://github.com/canonical/pebble/pull/434/commits/bc5d32d6acd44977d39c4f92b5290866707103e4